pennylane/hf/dipole.py

```diff
@@ -0,0 +1,310 @@
+            nd[2] = nd[2] + atomic_numbers[mol.symbols[i]] * mol.coordinates[i][2]
+
+        f = []
+        for i in range(3):
+            coeffs, ops = fermionic_one(constants[i], integrals[i], cutoff=cutoff)
+            f.append((anp.concatenate((nd[i], coeffs * (-1))), [[]] + ops))
+
+        return f
+
+    return _fermionic_dipole
+
+
+def dipole_moment(mol, cutoff=1.0e-12, core=None, active=None):
+    r"""Return a function that computes the qubit dipole moment observable.
+
+    Args:
+        mol (Molecule): the molecule object
+        cutoff (float): cutoff value for discarding the negligible dipole moment integrals
+        core (list[int]): indices of the core orbitals
+        active (list[int]): indices of the active orbitals
+
+    Returns:
+        function: function that computes the qubit dipole moment observable
+
+    **Example**
+
+    >>> symbols  = ['H', 'H']
+    >>> geometry = np.array([[0.0, 0.0, 0.0], [0.0, 0.0, 1.0]], requires_grad = False)
+    >>> alpha = np.array([[3.42525091, 0.62391373, 0.1688554],
+    >>>                   [3.42525091, 0.62391373, 0.1688554]], requires_grad=True)
+    >>> mol = qml.hf.Molecule(symbols, geometry, alpha=alpha)
+    >>> args = [alpha]
+    >>> dipole_moment(mol)(*args)[2].terms[1]
+    [PauliZ(wires=[0]),
+     PauliY(wires=[0]) @ PauliZ(wires=[1]) @ PauliY(wires=[2]),
+     PauliX(wires=[0]) @ PauliZ(wires=[1]) @ PauliX(wires=[2]),
+     PauliZ(wires=[1]),
+     PauliY(wires=[1]) @ PauliZ(wires=[2]) @ PauliY(wires=[3]),
+     PauliX(wires=[1]) @ PauliZ(wires=[2]) @ PauliX(wires=[3]),
+     PauliZ(wires=[2]),
+     PauliZ(wires=[3])]
+    """
+
+    def _dipole(*args):
+        r"""Compute the qubit dipole moment observable.
+
+        Args:
+            args (array[array[float]]): initial values of the differentiable parameters
+
+        Returns:
+            (list[Hamiltonian]): x, y and z components of the dipole moment observable
+        """
+        d = []
+        d_ferm = fermionic_dipole(mol, cutoff, core, active)(*args)
+        for i in d_ferm:
+            d.append(qubit_operator(i, cutoff=cutoff))
+
+        return d
+
+    return _dipole
+
+
+def fermionic_one(core_constant, integral, cutoff=1.0e-12):
+    r"""Create a fermionic operator from one-particle molecular orbital integrals.
+
+    Args:
+        core_constant (array[float]): the contribution of the core orbitals and nuclei
+        integral (array[float]): the one-particle molecular orbital integrals
+        cutoff (float): cutoff value for discarding the negligible integrals
+
+    Returns:
+        tuple(array[float], list[int]): fermionic coefficients and operators
+
+    **Example**
+
+    >>> constant = np.array([1.0])
+    >>> integral = np.array([[0.5, -0.8270995], [-0.8270995, 0.5]])
+    >>> coeffs, ops = fermionic_one(constant, integral)
+    >>> ops
+    [[], [0, 0], [0, 2], [1, 1], [1, 3], [2, 0], [2, 2], [3, 1], [3, 3]]
+    """
+    coeffs = anp.array([])
+
+    if core_constant != 0:
```

Comment:
`core_constant` is an array right? Can we compare it with `0` like this?

Reply:
It works with no problem but changed it to
`if core_constant != anp.array([0.0]):
`